src/otx/engine/engine.py

```diff
@@ -0,0 +1,486 @@
+        )
+        self.scheduler: LRSchedulerCallable = (
+            scheduler if scheduler is not None else self._auto_configurator.get_scheduler()
+        )
+
+    """
+    General OTX Entry Points
+    """
+
+    def train(
+        self,
+        max_epochs: int = 10,
+        seed: int | None = None,
+        deterministic: bool = False,
+        precision: _PRECISION_INPUT | None = "32",
+        val_check_interval: int | float | None = 1,
+        callbacks: list[Callback] | Callback | None = None,
+        logger: Logger | Iterable[Logger] | bool | None = None,
+        resume: bool = False,
+        **kwargs,
+    ) -> dict[str, Any]:
+        """Trains the model using the provided LightningModule and OTXDataModule.
+
+        Args:
+            max_epochs (int | None, optional): The maximum number of epochs. Defaults to None.
+            seed (int | None, optional): The random seed. Defaults to None.
+            deterministic (bool | None, optional): Whether to enable deterministic behavior. Defaults to False.
+            precision (_PRECISION_INPUT | None, optional): The precision of the model. Defaults to 32.
+            val_check_interval (int | float | None, optional): The validation check interval. Defaults to 1.
+            callbacks (list[Callback] | Callback | None, optional): The callbacks to be used during training.
+            logger (Logger | Iterable[Logger] | bool | None, optional): The logger(s) to be used. Defaults to None.
+            resume (bool, optional): If True, tries to resume training from existing checkpoint.
+            **kwargs: Additional keyword arguments for pl.Trainer configuration.
+
+        Returns:
+            dict[str, Any]: A dictionary containing the callback metrics from the trainer.
+
+        Example:
+        >>> engine.train(
+            max_epochs=3,
+            seed=1234,
+            deterministic=False,
+            precision="32",
+        )
+
+        CLI Usage:
+            1. you can train with data_root only. then OTX will provide default model.
+                ```python
+                otx train --data_root <DATASET_PATH>
+                ```
+            2. you can pick a model or datamodule as Config file or Class.
+                ```python
+                otx train
+                --data_root <DATASET_PATH>
+                --model <CONFIG | CLASS_PATH_OR_NAME> --data <CONFIG | CLASS_PATH_OR_NAME>
+                ```
+            3. Of course, you can override the various values with commands.
+                ```python
+                otx train
+                    --data_root <DATASET_PATH>
+                    --max_epochs <EPOCHS, int> --checkpoint <CKPT_PATH, str>
+                ```
+            4. If you have a complete configuration file, run it like this.
+                ```python
+                otx train --data_root <DATASET_PATH> --config <CONFIG_PATH, str>
+                ```
+        """
+        lit_module = self._build_lightning_module(
```

Comment:
If I understand it correctly, we always pass a torch (non-lightning) model to the engine and obtain the lightning module by wrapping the torch model in a subclass of `OTXLitModule`, depending on the task type. 

I'm wondering if this would provide sufficient flexibility to implement different model behaviours within a certain task type. Let's take anomaly models for example. We have different classes of anomaly models that each require a unique implementation of the training step and the test step (e.g. feature extraction based models vs. student-teacher models). I'm having some trouble seeing how we could implement these model-specific behaviours if we wrap all anomaly models in the same `OTXLitModule` subclass.

Reply:
I understand your example. I'll think more about the case where there are multiple LitModules per Task. Thanks for the nice comment.

Reply:
Can we collect Anomalib's LitModules into a single factory class called OTXAnomalibLitModule so that we can provide them for OTXModel?

What do you think? In the current OTX 2.0, there is currently only one LitModule class per Task.